src/main.cpp

```diff
@@ -225,8 +231,12 @@ int main(int argc, char* argv[])
         qWarning() << "Couldn't load font";
     }
 
-    Settings& settings = Settings::getInstance();
-    QString locale = settings.getTranslation();
+    settings = std::unique_ptr<Settings>(new Settings());
+    assert(settings);
```

Comment:
Afaik this assert will never trigger and the !settings will never evaluate to true since you're not passing `std::nothrow` to new. :P

Even if you're building with `-fno-exceptions`, new will (try to) throw if oom (and qTox will terminate.)

Reply:
Ah good point. Removed the check.